### ISSUE TITLE
Add formating twig code

### ISSUE THREAD

**Issue by Paviluf** [whatwedo/vscode-twig#15]
Hello,

Is it possible that you add formatting of twig code ?

Thanks !

**Comment by bgarrant** [comment]
We need this badly.  I tried tricking Beautify into thinking .twig as .html and that did not work.

**Comment by Paviluf** [comment]
https://github.com/whatwedo/vscode-twig/issues/10 is a duplicate (or the other way around)

**Comment by bgarrant** [comment]
I have no response to either thread. Is this easy to fix?

**Comment by Paviluf** [comment]
It seem that the extension is not developed anymore :(

**Comment by bgarrant** [comment]
Any idea what Twig extension we should use at this time? This one and the Twig Pack seem abandoned. Twig is such an important language used by many CMS these days. Can any convert the textmate bundle at https://github.com/Anomareh/PHP-Twig.tmbundle into a new extension?

**Comment by camposcristianeze** [comment]
I can't believe that I will change the IDE just for this. Is it hard to achive it? Any ideas/thoughts?

**Comment by bgarrant** [comment]
It is pretty easy to made the twig syntax highlighting and snippets from a TextMate bundle. See https://code.visualstudio.com/docs/extensions/themes-snippets-colorizers#_adding-a-new-language-colorizer. The issue is there is no Twig formatting (beautifier( that works with VSC at the moment.

**Comment by bgarrant** [comment]
@camposcristianeze My advice if you use twig...Get the PHPStorm EAP edition which is free to use.  You will never look back.

**Comment by bgarrant** [comment]
@camposcristianeze https://confluence.jetbrains.com/display/PhpStorm/PhpStorm+Early+Access+Program

**Comment by camposcristianeze** [comment]
Thanks @bgarrant . But yes, my issue is actually the formatting more than anything. For those who are interested, this extension is working:

https://marketplace.visualstudio.com/items?itemName=HexcodeTechnologies.vscode-prettydiff

**Comment by bgarrant** [comment]
@camposcristianeze That extension is formatting twig code?

**Comment by camposcristianeze** [comment]
Yes..But just discovered some issues in the formatting. Basically it's
changing the order of the tags? Something very strange.

**Comment by mblode** [comment]
I have finally solved the issue of formatting Twig in VSCode!

Over the past week I have put together a VSCode extension to include everything you would need in a Twig extension: https://marketplace.visualstudio.com/items?itemName=mblode.twig-language

This includes a WORKING Twig file formatter using PrettyDiff with a similar implementation as Atom-Beaitify. Moreover, my plugin includes syntax highlighting and language support, snippets and information on hover for Twig. Hopefully this fixes all of your issues with VSCode and Twig. If you run into a problem, simply open an issue on GitHub. 
Thanks and happy coding :)

**Comment by bgarrant** [comment]
Thanks Matt.  Works great
